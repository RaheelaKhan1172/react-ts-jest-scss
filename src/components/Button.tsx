import * as React from "react";
import { IButton } from "../interfaces/IButton";

export class Button extends React.Component<IButton, {}> {
 constructor(props: IButton) {
    super(props);

    this.onClick = this.onClick.bind(this);
  }

  onClick(event:any) {
    this.props.onClick(event); //call function sent from parent (on change)
  } 
  
  render() {
    return (
      <button
        className="tester"
        onClick={this.onClick}>
        Submit
      </button>
    );
  }
}
