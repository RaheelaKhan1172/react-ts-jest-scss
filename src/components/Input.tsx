import * as React from "react";

import { IInput } from "../interfaces/IInput";

export class Input extends React.Component<IInput, {}> {
  constructor(props: IInput) {
    super(props);

    this.onChange = this.onChange.bind(this); //'this' = undefined if not bound
  }

  onChange(event:any) {
    event.persist();
    this.props.onChange(event);
  }
  
  render() {
    return (
      <input className={this.props.classType? this.props.classType  : "inp"}
        type="text"
        value={this.props.value}
        name={this.props.name}
        onChange={this.onChange} />
    );
  }
}
