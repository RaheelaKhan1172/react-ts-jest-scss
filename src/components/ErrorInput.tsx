import * as React from "react";
import { IErrorInput } from "../interfaces/IErrorInput";
import { Input } from "./Input";

export class ErrorInput extends React.Component<IErrorInput, {}> {
  constructor(props: IErrorInput) {
    super(props);
  
    this.onChange = this.onChange.bind(this);
  }
  
  onChange(e:any) {
    e.persist();
    this.props.onChange(e);
  }

  render() {
    return (
      <div>
       <Input 
          classType="error"
          name={this.props.name}
          value={this.props.value}
          onChange={this.onChange} />
      Can't be blank, sorry
      </div>
    );
  }
}
