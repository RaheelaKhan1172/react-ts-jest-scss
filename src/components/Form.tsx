import * as React from "react";

import { FormState } from "../interfaces/IForm";
import { Button } from "./Button";
import { Input } from "./Input";
import { ErrorInput } from "./ErrorInput";

export class Form extends React.Component<{}, FormState> {
  //field, not prop
  public invalid:{email:boolean, password:boolean};

  constructor(props: any) {
    super(props);
    this.state = { email:"", password:"",invalid:false };

    this.invalid = {email:false, password:false}; 

    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);

  }

  onChange(event: any) {
    const name = event.target.name;
    this.setState((prevState) => ({
      [name]: event.target.value
    }));
  }

  onSubmit(event: any) {
    event.preventDefault();
    if (!this.state.email) {
      this.invalid.email = true;
    } else {
      this.invalid.email = false;
    } 
    if (!this.state.password) {
      this.invalid.password = true;
    } else {
      this.invalid.password = false;
    }

    if (this.invalid.password || this.invalid.email) {
      this.setState( () => ({
        invalid:true
      }));

    } else {
    this.setState( () => ({
      invalid:false
    }));
    }
  }

  render() {
    let passwordInput:any = null;
    let emailInput:any = null;

    if (this.state.invalid) {
      emailInput = ( this.invalid.email ? 
        <ErrorInput
            name="email"
            value={this.state.email}
            onChange={this.onChange} /> :
        <Input
          value={this.state.email}
          name="email"
          onChange={this.onChange} /> );
  
      passwordInput = ( this.invalid.password ? 
        <ErrorInput
          name="password"
          value={this.state.password}
          onChange={this.onChange} /> :
        <Input
          name="password"
          value={this.state.password}
          onChange={this.onChange} /> );
    } else {
      emailInput = <Input
          value={this.state.email}
          name="email"
          onChange={this.onChange} />;
      passwordInput = <Input
          name="password"
          value={this.state.password}
          onChange={this.onChange} />;
    }
    
    return (
      <div>
        <form className="form">
          <legend className="Fz(24px) C(#424242) Lh(20px) Mb(20px)"> Sign In </legend>
          <div className="textfield">
            { passwordInput }
            <label className="labels">Password</label>
          </div>
          <div className="textfield">
            { emailInput }
            <label className="labels">Email</label>
          </div>
          <Button
            onClick={this.onSubmit} />
       </form>
    </div>
    );
  }
}

