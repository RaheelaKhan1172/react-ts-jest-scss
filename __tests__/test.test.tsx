import * as React from "react";
import * as TestUtils from "react-addons-test-utils";
import renderer from 'react-test-renderer';

import { Form } from "../src/components/Form";

describe("<Form />", () => {
  it("renders", () => {
    expect(TestUtils.createRenderer().render(
      <Form />  
    )).toMatchSnapshot();
  });
});

