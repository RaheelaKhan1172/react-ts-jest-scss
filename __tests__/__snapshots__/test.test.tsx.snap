// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders 1`] = `
<div>
  <form
    className="form"
  >
    <legend
      className="Fz(24px) C(#424242) Lh(20px) Mb(20px)"
    >
       Sign In 
    </legend>
    <div
      className="textfield"
    >
      <Input
        name="password"
        onChange={[Function]}
        value=""
      />
      <label
        className="labels"
      >
        Password
      </label>
    </div>
    <div
      className="textfield"
    >
      <Input
        name="email"
        onChange={[Function]}
        value=""
      />
      <label
        className="labels"
      >
        Email
      </label>
    </div>
    <Button
      onClick={[Function]}
    />
  </form>
</div>
`;
